### PR TITLE
fix: visibility of update password & security fields using identityprovider

### LINKED ISSUE #690: Bug: The 'Update password' should not be visible or accessible for OAuth signup/login.
### Issue Description

Currently, even if someone is signed up or logged in through Google, the 'update password' field is still visible to them.

### Expected Behavior

The 'update password' field should only be visible to users who have signed up using email and password. For OAuth users, display a message in the update password section indicating that password updates are not applicable or directing them to the respective OAuth provider for security management.

Screenshot 2023-11-27 235730

### Current Behavior

_No response_

### Screenshots (optional)

_No response_

### Operating System [e.g., Windows 10]

_No response_

### Browser [e.g., Chrome, Firefox]

_No response_

### Version [e.g., 2.0.1]

_No response_

### Please check the boxes that apply to this issue report.

- [ ] I have searched the existing issues to make sure this is not a duplicate.
- [ ] I have provided steps to reproduce the issue.
- [ ] I have included relevant environment information.
- [ ] I have included any relevant screenshots.
- [ ] I understand that this is a voluntary contribution and that there is no guarantee of resolution.
- [ ] I want to work on creating a PR for this issue if approved

### PR DESCRIPTION
**Pull request by anikdhabal**
fixes #690 
Screenshot 2023-12-01 233708

### CODE REVIEW

**Comment by anikdhabal** [comment]
cc @Mythie @catalinpit

**Comment by anikdhabal** [comment]
cc @catalinpit @Mythie

File: apps/web/src/app/(dashboard)/settings/security/page.tsx [hunk 1]
```diff
@@ -17,28 +18,42 @@ export default async function SecuritySettingsPage() {
...
+            </div>
+          )}
+        </div>
+      ) : (
+        <div>
+          <h4 className="text-lg font-medium">
+            Your account is managed by {identityProviderName[user.identityProvider]}
+          </h4>
+          <p className="text-muted-foreground mt-2 text-sm">
+            To change your password, enable two-factor authentication and more, please visit your{' '}
```
Comment (catalinpit) [inline apps/web/src/app/(dashboard)/settings/security/page.tsx:54]: This text is not very clear. Could you update it, please?

Perhaps, something like `To change your password, enable two-factor authentication and then visit your{' '}....`

Also, did you test this change locally?

Comment (anikdhabal) [reply apps/web/src/app/(dashboard)/settings/security/page.tsx:54]: Yes, I've test this locally. Let me update the text.

Comment (anikdhabal) [reply apps/web/src/app/(dashboard)/settings/security/page.tsx]: Update it and test the changes locally.

Comment (anikdhabal) [reply apps/web/src/app/(dashboard)/settings/security/page.tsx]: cc @catalinpit

**Review by dguyen — APPROVED** [review]
Thanks for this, looks good to me :)

**Pull request merged**